src/data_c.c4

```diff
@@ -439,6 +479,33 @@ SHMEM_PROF_DEF_CTX_PUT_N_SIGNAL_NBI(`mem')
     }                                                         \
   }
 
+#define SHMEM_DEF_IBPUT(STYPE,TYPE)                           \
+  void SHMEM_FUNCTION_ATTRIBUTES                              \
+  SHMEMX_FUNC_PROTOTYPE(STYPE##_ibput, TYPE *target,            \
+                       const TYPE *source, ptrdiff_t tst,     \
+                       ptrdiff_t sst, size_t bsize, size_t nblocks, int pe)  \
+    SHMEM_ERR_CHECK_INITIALIZED();                            \
+    SHMEM_ERR_CHECK_PE(pe);                                   \
+    SHMEM_ERR_CHECK_CTX(ctx);                                 \
+    SHMEM_ERR_CHECK_POSITIVE(tst);                            \
+    SHMEM_ERR_CHECK_POSITIVE(sst);                            \
+    SHMEM_ERR_CHECK_SYMMETRIC(target, sizeof(TYPE) * ((nblocks-1) * tst + 1)); \
+    SHMEM_ERR_CHECK_NULL(source, nblocks);                    \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(tst, bsize);             \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(sst, bsize);             \
```

Comment:
Should we also check for `bsize >= 0`? It needs to be, because otherwise the `nblocks` loop below will run needlessly. May be, we should add that in the spec as well. 
If we add that check then line num 490 and 491 becomes redundant.

Reply:
That's a good idea. I'll add that check for bsize and remove the checks that then become redundant. bsize should be strictly greater than 0, right?

Reply:
I guess you can provide a value of 0 to bsize, but then isn't the call essentially a no-op?

Reply:
For error checking, you can add the check if `bsize > 0`. But, not sure if we should generate an error for `bsize = 0`. May be we can have that check at the beginning and skip all.

Reply:
I think it might be ok to accept passing `bsize == 0`...  SOS never returns from the other RMA routines when `nelems` is zero, right?  I think our strategy has been to pass zero-byte messages all the way through libfabric, where it's documented as a valid transfer (see [fi_rma](https://ofiwg.github.io/libfabric/main/man/fi_rma.3.html)).  But yeah, if `nblocks` is big then that's a pretty weird use-case ;)